### DISCUSSION TITLE
FP absolute links vs relative links explanation

### DISCUSSION THREAD

**Discussion by Illegit** [freeplane/freeplane#863] [Help]
May you provide a review of the characteristics of relative links versus absolute links?
First characteristics and after usage, if possible.
I need to refresh my understanding of how to apply them, at now to be sure in using some addons as MDI I utilize always absolute links, but what about using MDI with relative links?
Many thanks

**Comment by EdoFro** [comment]
When I created the MDI addon, I had problems understanding relative paths and how to code the addon to work with both kinds of links. I decided to make it work with absolute links first. I got used to absolute links, because I don't share my maps with anybody and when I move a map I don't need to move all the files with it or to redefine all the links.
But theoretically, MDI should be able to handle relative links with just minor adjustments.

Maybe someday.

**Comment by Illegit** [comment]
Thank you. Appreciate your answer. MDI is absolutely fundamental, I use it daily. I'll keep absolute links.

**Reply by EdoFro** [reply]
Hi @Illegit,
@bepolymathe had some issues with MDI (you can see them [here](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/issues/1 ))
So I decided to pack MDI into an Add-On.

Can you please test it too, so I can get all the bugs at once.

thanks,

edo

**Reply by EdoFro** [reply]
look at the issue thread from [this comment](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/issues/1#issuecomment-1384537093) to the end

**Comment by Illegit** [comment]
As the above, one topic is how to use MDI, but still, I need to identify a place where to read how to use relative links vs absolute ones, any suggestions? Thank s.

**Comment by Illegit** [comment]
Today I tested v0. 0.5 and it worked perfectly. Nice.
I will test again in this weekend and feedback.
Thanks

**Reply by EdoFro** [reply]
Hi @Illegit,
I released a new test version that should work with absolute and/or relative links.
look here: [MDI discussion "v0.0.7"](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/discussions/19)
please test it with new maps first

**Comment by Illegit** [comment]
Hi Edofro, I applied version 9 on 2 different PC's, in 1 it is working perfectly but in another, I got the problem please have a look at the attached zip file
[FP MDI9 topics abs links.zip](https://github.com/freeplane/freeplane/files/10714723/FP.MDI9.topics.abs.links.zip)

**Reply by EdoFro** [reply]
¿did you delete the MDI.groovy file from the <user directory>/lib folder? (you have to)

This file was used in the previous versions, but now it get installed as a jar file by the add-on.

**Reply by EdoFro** [reply]
I think you have to uninstall the addon and also [uninstall any other MDI script (from the really old version)](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/issues/1#issuecomment-1385856088) before reinstalling the last version from MDI

**Comment by Illegit** [comment]
Thanks I review it


Il lun 13 feb 2023, 01:49 EdoFro ***@***.***> ha scritto:

**Comment by Illegit** [comment]
<img width="269" alt="Screenshot 2023-02-13 215444" src="https://user-images.githubusercontent.com/62674157/218581051-1b76a60a-17fc-45e3-bc3c-d5a02e8a29f8.png">
Hi EdoFro, I uninstall the previous MDI and eliminated all of my MDI.groovy from my pc, then I installed the mdi v09 addon, but I got the above error, so I uninstall the v09 and reinstalled the v05 and the addon is running correctly for abs links (every step I close and open FP), What I'm missing? Thanks
illegit

**Reply by EdoFro** [reply]
can you add a new zip file with your log file? (with installed v09 addon)

Maybe you can create an issue in https://github.com/EdoFro/Freeplane_Map-Drive-Inator/issues
